mongo_connector/doc_managers/neo4j_doc_manager.py

```diff
@@ -67,7 +67,19 @@ def upsert(self, doc, namespace, timestamp):
 
   def bulk_upsert(self, docs, namespace, timestamp):
     """Insert multiple documents into Neo4j."""
-    LOG.error("Bulk")
+    metadata = { "ns": namespace, "_ts": timestamp }
+    tx = self.graph.cypher.begin()
+    for doc in docs:
+      index, doc_type = self._index_and_mapping(namespace)
+      doc_id = u(doc.pop("_id"))
+      doc = self._formatter.format_document(doc)
+      builder = NodesAndRelationshipsBuilder(doc, doc_type, doc_id)
+      self.apply_id_constraint(builder.doc_types)
+      for statement in builder.query_nodes.keys():
+        tx.append(statement, builder.query_nodes[statement])
+      for relationship in builder.relationships_query.keys():
+        tx.append(relationship, builder.relationships_query[relationship])
+    tx.commit()
```

Comment:
We should make sure that the number of statements in the transaction does not become excessive. Given that we don't know the number of updates per statement, I think the transaction should be committed once 5000 statements are appended. I would suggest keeping a counter of the number of statements added to the transaction and calling `tx.commit()` when there are 5000 statements in the transaction.